packages/editor/src/components/block-actions/index.js

```diff
@@ -107,6 +110,11 @@ export default compose( [
 					insertDefaultBlock( {}, rootClientId, lastSelectedIndex + 1 );
 				}
 			},
+			onToggleEditor() {
+				if ( ! isLocked ) {
```

Comment:
It looks like a user also shouldn't be able to toggle when the block is reusable, so it'd be good to add another condition here to cover that.